http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/app/AppInstall.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.dsde.workbench.leonardo.app
+
+import cats.mtl.Ask
+import org.broadinstitute.dsde.workbench.azure.AzureCloudContext
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.ServiceAccountName
+import org.broadinstitute.dsde.workbench.leonardo.dao.StorageContainerResponse
+import org.broadinstitute.dsde.workbench.leonardo.util.AKSInterpreterConfig
+import org.broadinstitute.dsde.workbench.leonardo.{
+  App,
+  AppContext,
+  AppType,
+  LandingZoneResources,
+  ManagedIdentityName,
+  WorkspaceId
+}
+import org.broadinstitute.dsp.Values
+import org.http4s.Uri
+import org.http4s.headers.Authorization
+
+/**
+ * Defines how to install a Kubernetes App.
+ */
+trait AppInstall[F[_]] {
+
+  /** List of WSM-controlled databases the app requires. */
+  def databases: List[Database]
+
+  /** Builds helm values to be passed to the app. */
+  def buildHelmOverrideValues(params: BuildHelmOverrideValuesParams)(implicit ev: Ask[F, AppContext]): F[Values]
+
+  /** Checks status of the app. */
+  def checkStatus(baseUri: Uri, authHeader: Authorization)(implicit ev: Ask[F, AppContext]): F[Boolean]
+}
+
+object AppInstall {
+
+  /** Maps AppType to AppInstall. */
+  def appTypeToAppInstall[F[_]](wdsAppInstall: WdsAppInstall[F],
```

Comment:
feels to me this shouldn't be needed. Comipler should be able to find the correct instance depending on the type 🤔 ... can defer to future improvements tho

Reply:
oh I see..you want the extension methods on `AppType`... 👍